### PR TITLE
add history

### PR DESCRIPTION
**Pull request by ndoriomedov**
http://outsourcing.nat.tepkom.ru:20080/redmine/projects/checkrightsservice

### CODE REVIEW

File: Common.Configurations.targets [hunk 1]
```diff
@@ -26,17 +26,18 @@
   </PropertyGroup>
 
   <ItemGroup Condition="($(AssemblyName.Contains('.Provider')) Or $(AssemblyName.Contains('.Models.Db'))) And $(AssemblyName.Contains('.UnitTests')) != 'true'">
-    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="6.0.0" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.InMemory" Version="6.0.0" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.Relational" Version="6.0.0" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="6.0.0" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="6.0.5" />
```
Comment (MariyaSin) [inline Common.Configurations.targets:29]: add kernel.EFSupport nuget v 1.0.1 and remove the references

File: src/RightsService.Broker/Consumers/CreateUserRoleConsumer.cs [hunk 2]
```diff
@@ -58,9 +58,11 @@ private async Task UpdateCacheAsync(Guid userId, Guid roleId)
...
     public async Task Consume(ConsumeContext<ICreateUserRolePublish> context)
     {
-      if (await _roleRepository.DoesExistAsync(context.Message.RoleId))
+      if (await _roleRepository.DoesExistAsync(context.Message.RoleId)
+        && !await _userRoleRepository.DoesExistAsync(context.Message.UserId))
       {
         await _userRoleRepository.CreateAsync(_mapper.Map(context.Message));
+
         await UpdateCacheAsync(context.Message.UserId, context.Message.RoleId);
       }
```
Comment (MariyaSin) [inline src/RightsService.Broker/Consumers/CreateUserRoleConsumer.cs:67]: add log with error

File: src/RightsService.Business/Commands/Role/FindRolesCommand.cs [hunk 3]
```diff
@@ -99,6 +99,7 @@ public async Task<FindResultResponse<RoleInfo>> ExecuteAsync(FindRolesFilter fil
 
       FindResultResponse<RoleInfo> result = new();
 
+      //TODO - find using one method
```
Comment (MariyaSin) [inline src/RightsService.Business/Commands/Role/FindRolesCommand.cs:102]: ?

File: src/RightsService.Business/Commands/Role/FindRolesCommand.cs [hunk 4]
```diff
@@ -99,6 +99,7 @@ public async Task<FindResultResponse<RoleInfo>> ExecuteAsync(FindRolesFilter fil
```
Comment (MariyaSin) [inline src/RightsService.Business/Commands/Role/FindRolesCommand.cs:99]: use response creator 91-96 lines

File: src/RightsService.Business/Commands/Role/FindRolesCommand.cs [hunk 5]
```diff
@@ -99,6 +99,7 @@ public async Task<FindResultResponse<RoleInfo>> ExecuteAsync(FindRolesFilter fil
 
       FindResultResponse<RoleInfo> result = new();
 
+      //TODO - find using one method
       (List<(DbRole role, List<DbRightLocalization> rights)> roles, int totalCount) = filter.IncludeDeactivated
         ? await _roleRepository.FindAllAsync(filter)
```
Comment (MariyaSin) [inline src/RightsService.Business/Commands/Role/FindRolesCommand.cs:104]: why you do not filter roles in repo?

File: src/RightsService.Business/Commands/User/EditUserRoleCommand.cs [hunk 6]
```diff
@@ -56,17 +56,24 @@ public async Task<OperationResultResponse<bool>> ExecuteAsync(EditUserRoleReques
...
+      }
+
+      if (!request.RoleId.HasValue)
+      {
+        response.Body = await _repository.RemoveAsync(request.UserId);
+      }
+      else
+      {
+        response.Body = oldUser is not null
+          ? await _repository.EditAsync(oldUser, _mapper.Map(request))
```
Comment (MariyaSin) [inline src/RightsService.Business/Commands/User/EditUserRoleCommand.cs:73]: do you need map request?

File: src/RightsService.Data/UserRoleRepository.cs [hunk 7]
```diff
@@ -102,19 +118,23 @@ public async Task<List<DbUserRole>> GetWithRightsAsync()
 
     public async Task<bool> RemoveAsync(Guid userId)
     {
-      DbUserRole user = await _provider.UsersRoles.FirstOrDefaultAsync(u => u.UserId == userId && u.IsActive);
+      DbUserRole user = await _provider.UsersRoles.FirstOrDefaultAsync(u => u.UserId == userId);
```
Comment (MariyaSin) [inline src/RightsService.Data/UserRoleRepository.cs:121]: may be return "&& u.IsActive"? this protects against re-writing to the history table if your method is called not from your command but from another place

**Review by MariyaSin — APPROVED** [review]

**Review by VLiamin — APPROVED** [review]

**Pull request merged**